### PR TITLE
Refactor cli.

### PR DESCRIPTION
**Pull request by mshima**
Closes https://github.com/jhipster/generator-jhipster/issues/11896.

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: test/cli/jdl-fork.spec.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                    '--skip-install',
+                    '--fork',
+                    '--skip-sample-repository',
+                ]);
+        });
+        after(() => {
+            sandbox.restore();
+        });
+
+        it('should succeed', function () {
```
Comment (qmonmert) [inline test/cli/jdl-fork.spec.js:90]: `function () {` => `() => {` 😇

Comment (mshima) [reply test/cli/jdl-fork.spec.js:90]: Not possible due to `this.timeout(60000);`.

Comment (qmonmert) [reply test/cli/jdl-fork.spec.js:90]: ok my bad

Comment (mshima) [reply test/cli/jdl-fork.spec.js:90]: no problem.

**Comment by pascalgrimaud** [comment]
Trusting you here @mshima !

**Pull request merged**